app/models/dialog/terraform_template_service_dialog.rb

```diff
@@ -0,0 +1,92 @@
+require "JSON"
+
+class Dialog
+  class TerraformTemplateServiceDialog
+    def self.create_dialog(label, terraform_template, extra_vars)
+      new.create_dialog(label, terraform_template, extra_vars)
+    end
+
+    # This dialog is to be used by a terraform template service item
+    def create_dialog(label, terraform_template, extra_vars)
+      Dialog.new(:label => label, :buttons => "submit,cancel").tap do |dialog|
+        tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+        if terraform_template.present?
+          add_template_variables_group(tab, 0, terraform_template)
+        end
+        if extra_vars.present?
+          add_variables_group(tab, 1, extra_vars)
+        end
+        dialog.save!
+      end
+    end
+
+    private
+
+    def add_template_variables_group(tab, position, terraform_template)
```

Comment:
Is position necessary as an argument?  This only ever seems to be called with `position=0`, likewise `add_variables_group` only ever called with `position=1`

Reply:
Improved the code,  though not removing position argument itself.
the first group always start at '0',  
next group will may start at '1', if there are terraform template variables 
else next group will start at '0', if there are NO terraform template variables